src/auxil.c

```diff
@@ -287,146 +291,147 @@ c_float compute_pri_tol(OSQPWorkspace *work, c_float eps_abs, c_float eps_rel) {
-    temp_rel_eps = vec_norm_inf(work->Aty, work->data->n);
+    temp_rel_eps = OSQPVectorf_norm_inf(work->Aty);
     max_rel_eps  = c_max(max_rel_eps, temp_rel_eps);
 
     // ||P*x||
-    temp_rel_eps = vec_norm_inf(work->Px, work->data->n);
+    temp_rel_eps = OSQPVectorf_norm_inf(work->Px);
     max_rel_eps  = c_max(max_rel_eps, temp_rel_eps);
   }
 
   // eps_dual
   return eps_abs + eps_rel * max_rel_eps;
 }
 
-c_int is_primal_infeasible(OSQPWorkspace *work, c_float eps_prim_inf) {
+c_int is_primal_infeasible(OSQPSolver *solver, c_float eps_prim_inf) {
+
   // This function checks for the primal infeasibility termination criteria.
   //
   // 1) A' * delta_y < eps * ||delta_y||
   //
   // 2) u'*max(delta_y, 0) + l'*min(delta_y, 0) < -eps * ||delta_y||
   //
 
-  c_int i; // Index for loops
   c_float norm_delta_y;
   c_float ineq_lhs = 0.0;
+  OSQPSettings*  settings = solver->settings;
+  OSQPWorkspace* work     = solver->work;
 
   // Project delta_y onto the polar of the recession cone of [l,u]
-  for (i = 0; i < work->data->m; i++) {
-    if (work->data->u[i] > OSQP_INFTY * MIN_SCALING) {          // Infinite upper bound
-      if (work->data->l[i] < -OSQP_INFTY * MIN_SCALING) {       // Infinite lower bound
-        // Both bounds infinite
-        work->delta_y[i] = 0.0;
-      } else {
-        // Only upper bound infinite
-        work->delta_y[i] = c_min(work->delta_y[i], 0.0);
-      }
-    } else if (work->data->l[i] < -OSQP_INFTY * MIN_SCALING) {  // Infinite lower bound
-      // Only lower bound infinite
-      work->delta_y[i] = c_max(work->delta_y[i], 0.0);
-    }
-  }
+  project_polar_reccone(work->delta_y,
+                        work->data->l,
+                        work->data->u,
+                        OSQP_INFTY * MIN_SCALING);
 
   // Compute infinity norm of delta_y (unscale if necessary)
-  if (work->settings->scaling && !work->settings->scaled_termination) {
+  if (settings->scaling && !settings->scaled_termination) {
     // Use work->Adelta_x as temporary vector
-    vec_ew_prod(work->scaling->E, work->delta_y, work->Adelta_x, work->data->m);
-    norm_delta_y = vec_norm_inf(work->Adelta_x, work->data->m);
-  } else {
-    norm_delta_y = vec_norm_inf(work->delta_y, work->data->m);
+    OSQPVectorf_ew_prod(work->Adelta_x,
+                        work->scaling->E,
+                        work->delta_y);
+    norm_delta_y = OSQPVectorf_norm_inf(work->Adelta_x);
+  }
+  else {
+    norm_delta_y = OSQPVectorf_norm_inf(work->delta_y);
   }
 
   if (norm_delta_y > eps_prim_inf) { // ||delta_y|| > 0
 
-    for (i = 0; i < work->data->m; i++) {
-      ineq_lhs += work->data->u[i] * c_max(work->delta_y[i], 0) + \
-                  work->data->l[i] * c_min(work->delta_y[i], 0);
-    }
+    ineq_lhs  = OSQPVectorf_dot_prod_signed(work->data->u, work->delta_y,+1);
+    ineq_lhs += OSQPVectorf_dot_prod_signed(work->data->l, work->delta_y,-1);
 
     // Check if the condition is satisfied: ineq_lhs < -eps
```

Comment:
Haven't we discussed that we should relax this inequality, i.e. replace `ineq_lhs < -eps` with `ineq_lhs < eps`? See Eqs. (30) and (31) [here](https://stanford.edu/~boyd/papers/pdf/admm_infeas.pdf).

A similar comment holds for `q'*delta_x < eps`.

The idea is that if inequalities hold for a given `eps`, then they should also hold for a higher value of `eps`.

Reply:
I agree, but I'm reluctant to change it in this PR.   I think for this it would be better to implement it as an isolated change so that we can properly test the effect on some infeasible benchmarks.

Reply:
Moved to #180.